views/default/navigation/menu/user_hover/contents.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+// g = guid, pog = page_owner_guid, c = contexts, m = mac
+$guid = (int) get_input("g", 0, false);
+$page_owner_guid = (int) get_input("pog", 0, false);
+$contexts = (array) get_input("c", array(), false);
+$mac = get_input("m", "", false);
+$input = (array) get_input("i", array(), false);
+
+// verify MAC
+$data = array($guid, $page_owner_guid, $contexts, $input);
+
+if (!elgg_build_hmac($data)->matchesToken($mac)) {
```

Comment:
This won't work because array arguments will be converted to "Array". elgg_build_hmac() should probably throw in that case. Easiest way to get this working:
```php
$data = serialize([$guid, $page_owner_guidm $contexts, $input]);
```